code/__DEFINES/preferences.dm

```diff
@@ -1,28 +1,55 @@
 
 //Preference toggles
-#define SOUND_ADMINHELP			(1<<0)
-#define SOUND_MIDI				(1<<1)
-#define SOUND_AMBIENCE			(1<<2)
-#define SOUND_LOBBY				(1<<3)
-#define MEMBER_PUBLIC			(1<<4)
-#define INTENT_STYLE			(1<<5)
-#define MIDROUND_ANTAG			(1<<6)
-#define SOUND_INSTRUMENTS		(1<<7)
-#define SOUND_SHIP_AMBIENCE		(1<<8)
-#define SOUND_PRAYERS			(1<<9)
-#define ANNOUNCE_LOGIN			(1<<10)
-#define SOUND_ANNOUNCEMENTS		(1<<11)
-#define DISABLE_DEATHRATTLE		(1<<12)
-#define DISABLE_ARRIVALRATTLE	(1<<13)
-#define COMBOHUD_LIGHTING		(1<<14)
-
-#define DEADMIN_ALWAYS			(1<<15)
-#define DEADMIN_ANTAGONIST		(1<<16)
-#define DEADMIN_POSITION_HEAD	(1<<17)
-#define DEADMIN_POSITION_SECURITY	(1<<18)
-#define DEADMIN_POSITION_SILICON	(1<<19)
-
-#define TOGGLES_DEFAULT (SOUND_ADMINHELP|SOUND_MIDI|SOUND_AMBIENCE|SOUND_LOBBY|MEMBER_PUBLIC|INTENT_STYLE|MIDROUND_ANTAG|SOUND_INSTRUMENTS|SOUND_SHIP_AMBIENCE|SOUND_PRAYERS|SOUND_ANNOUNCEMENTS)
+#define PREFTOGGLE_SOUND_ADMINHELP				(1<<0)
```

Comment:
Crazy idea, but hear me out: What if you get rid of all of these (from this file at least)?

Specifically, what if you leave all the individual booleans in /datum/preferences, as well as breaking out the exsting packed toggles into their own fields? The wider code would just interact with these variables, and only the save/load code would need to know about packing them into bitfields.

Advantages:
- Less macros in the code, which hinder understanding because you need to cross-ref them
- Slightly faster, because you're not doing masking every time you check a pref - it's a one-time cost only, when loading/saving preferences.
- It hides the implementation details where they belong. If the storage format changes in the future, it won't require changing all the code. (And it will make this current change much smaller.)
- Most importantly, it reduces the chance of mistakes:
```
if (usr.client.pref.toggles & PREFTOGGLE_2_FEATURE)
    DoCoolStuff()
```
is a mistake that's easy to miss in review *and* testing.

Disadvantages:
- Requires more space in memory. I don't know exactly how much space fields take up, but I'm going to assume 8 bytes for a bool, just to be generous. 8 * 50 toggle prefs * 100 clients = 40000 bytes, or 40K, i.e. it's tiny.
- More work for you.

Reply:
You would save like 1 clock cycle in performance, but the hiding implantation point is good

Reply:
I would much prefer it to stay as bitflags, mainly to reduce datum bloat. Having 35+ extra booleans tacked onto the prefs datum will not only make it even more painful to debug and VV, but we use bitflags for a reason, hence why object flags are, well, flags, as opposed to `does_conduct`, `no_deconstruct`, etc. It also means we can wrap defaults as a single int much better as opposed to a ton of var sets.

Reply:
It's interesting that you bring up debugging and VV, because for me, having broken-out and well named fields is much easier to debug and VV than a packed bitfield, where you need to bring up the source to decode what the value means. Unless you write a custom VV hook, but then you're throwing even more code at the problem...

But if it were just a single bitfield, I wouldn't have said anything. Although I think broken-out fields are generally better, they're not _that_ much better, and there are also times when they make the most sense because the code needs to pass around their values as a single unit. The reason I spoke up for this one is the risk of having bugs becomes much greater when the constants are spread across multiple bitfields, and you can end up applying the right mask to the wrong field.